### ISSUE TITLE
Trail ends abruptly with no message to user

### ISSUE THREAD

**Issue by MdBee** [exercism/exercism.io#2358]
Hi.  I just submitted Grains for Swift.  When I tried to fetch the next exercise I received Point Mutations for Scheme.  Is there a version for Swift?  Thanks for everything.

**Comment by Dog** [comment]
Point Mutations has not been implemented in Swift, and it looks like grains is the last exercise at the moment. I believe that another exercise was pulled is a bug.

Congratulations on finishing them all.

If you want more problems, the Exercism repository for Swift is here:
https://github.com/exercism/xswift

You could help implement more exercises if you'd like, and help contribute more problems for the swift track on the site. The repository with metadata for all problems is here:
https://github.com/exercism/x-common

**Comment by kytrinyx** [comment]
When you fetched, did you say `exercism fetch swift`? Did `point-mutations` end up in the `swift/` directory?

**Comment by MdBee** [comment]
Hi Katrina,
I said
exercism fetch
from within the swift/grains/ directory.
Point-Mutations appeared in the scheme/ directory.
Thanks for helping my coding!
Matt

415-233-3307

**Comment by kytrinyx** [comment]
OK, cool, so it's not _technically_ a bug, but it's definitely confusing, and a terrible UX experience.

We have an open issue to address it, but we haven't decided _how_ yet, which is why it's stalled. Time to brainstorm some actual approaches! https://github.com/exercism/cli/issues/53

**Comment by kytrinyx** [comment]
OK, the status of this ticket is that we're implementing a new version of the x-api (https://github.com/exercism/x-api/issues/75), which we will use to implement a new endpoint on the exercism.io api, which the CLI can call. This endpoint will call x-api to get the exercises, and then it will check whether or not the track is completed (comparing the track config to the list of exercises submitted on that track).

So, in other words, we are going to need to:

- implement the endpoint in x-api v3 to fetch the exercises
- implement the endpoint in exercism which will get the exercises from x-api and figure out if the track is completed
- reimplement the `fetch` logic within the CLI to use the new exercism endpoint.

**Comment by kytrinyx** [comment]
Since we have other open tickets for this I'm going to close this issue.
